tensorflow/core/kernels/cudnn_rnn_ops.cc

```diff
@@ -1074,7 +1148,33 @@ class CudnnRNNParamsToCanonical<GPUDevice, T> : public CudnnRNNKernelCommon {
  public:
   explicit CudnnRNNParamsToCanonical(OpKernelConstruction* context)
       : CudnnRNNKernelCommon(context) {
-    OP_REQUIRES_OK(context, context->GetAttr("num_params", &num_params_));
+    if (context->HasAttr("num_params")) {
```

Comment:
Since it used to enforce that attribute num_params exists, should we ensure that one and only one set of attributes {num_params} and {num_params_weights, num_params_biases} is set?

Reply:
I tried to only use {num_params_weights, num_params_biases}. I use the existence of `num_params` to correctly set num_params_weights and num_params_biases (at the end of this constructor). In the following `Compute()`, only num_params_weights and num_params_biases are used.